src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -524,7 +524,7 @@ var ReactCompositeComponentMixin = {
 
     // Distinguish between a props update versus a simple state update
     if (prevParentElement === nextParentElement) {
-      nextContext = inst.context;
+      nextContext = this._processContext(nextUnmaskedContext);
```

Comment:
Yeah, that's the test in this diff, right?  Unless you changed it since I copy-pasted it.  Test passes without this line.